api/app/controllers/spree/api/v2/storefront/cart_controller.rb

```diff
@@ -27,14 +27,18 @@ def add_item
               spree_authorize! :update, spree_current_order, order_token
               spree_authorize! :show, variant
 
-              dependencies[:add_item_to_cart].call(
+              result = dependencies[:add_item_to_cart].call(
                 order: spree_current_order,
                 variant: variant,
                 quantity: params[:quantity],
                 options: params[:options]
               )
 
-              render_serialized_payload serialized_current_order, 200
+              if result.success?
+                render_serialized_payload serialized_current_order, 200
```

Comment:
I suppose we can omit 200 since it's used as a fallback in `render_serialized_payload` method?